metadata-mapping/src/main/java/com/iluwatar/metamapping/service/UserService.java

```diff
@@ -0,0 +1,139 @@
+package com.iluwatar.metamapping.service;
+
+import com.iluwatar.metamapping.model.User;
+import com.iluwatar.metamapping.utils.HibernateUtil;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+/**
+ * Service layer for user.
+ */
+@Slf4j
+public class UserService {
+  private static final SessionFactory factory = HibernateUtil.getSessionFactory();
+
+  /**
+   * List all users.
+   * @return list of users
+   */
+  public List<User> listUser() {
+    LOGGER.info("list all users.");
+    Session session = factory.openSession();
+    Transaction tx = null;
+    List<User> users = new ArrayList<>();
+    try {
```

Comment:
Maybe we could use try-with-resources to close the session automatically? Check elsewhere too.